lib/Conversion/P4HIRToBMv2IR/P4HIRToBMv2IR.cpp

```diff
@@ -21,7 +32,211 @@ namespace P4::P4MLIR {
 using namespace P4::P4MLIR;
 
 namespace {
+
+BMv2IR::FieldInfo convertFieldInfo(P4HIR::FieldInfo p4Field) {
+    return BMv2IR::FieldInfo(p4Field.name, p4Field.type);
+}
+
+struct P4HIRToBMv2IRTypeConverter : public mlir::TypeConverter {
+    P4HIRToBMv2IRTypeConverter() {
+        addConversion([&](mlir::Type t) { return t; });
+        addConversion([&](P4HIR::HeaderType headerType) {
+            SmallVector<BMv2IR::FieldInfo> newFields;
```

Comment:
Do you want to enforce that fields here are assumed to be `simple`, e.g. no structs, etc.?